src/main/java/justin/Parser.java

```diff
@@ -0,0 +1,76 @@
+package justin;
+
+import justin.JustinException;
+
+public class Parser {
+
+    private String fullText;
+
+    public Parser(String fullText) {
+        this.fullText = fullText;
+    }
+
+    public String checkCommand() throws JustinException {
+        try {
+
+            validate(fullText);
+
+            if (fullText.equals("bye")) {
+
+                return "BYE";
+
+            } else if (fullText.equals("list")) {
+
+                return "LIST";
+
+            } else if (fullText.contains("done")) {
+
+                return "DONE";
+
+            } else if (fullText.contains("deadline")) {
+
+                return "DEADLINE";
+
+            } else if (fullText.contains("todo")) {
+
+                return "TODO";
+
+            } else if (fullText.contains("event")) {
+
+                return "EVENT";
+
+            }
+            else if (fullText.contains("delete")) {
+
+                return "DELETE";
+
+            } else if (fullText.contains("find")) {
+
+                return "FIND";
+
+            }
+
+            else {
+
+                return "ADD";
+
+            }
+        } catch (JustinException m) {
+            throw new JustinException(m.getMessage());
+        }
+    }
+
+    static void validate(String text) throws JustinException {
+        if (text.length() < 5 && text.contains("todo") ) { // case 1
+            throw new JustinException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        else if (text.contains("blah")) { // case 2
```

Comment:
According to convention and for better readability, maybe consider indenting the trailing comments to the same tab setting?